app/representers/api/v1/exercise_representer.rb

```diff
@@ -5,7 +5,7 @@ class ExerciseRepresenter < Roar::Decorator
 
     # Attachments may (for a while) contain collaborator solution attachments, so
     # only show them to those who can see solutions
-    has_attachments(if: lambda { |args| can_view_solutions?(args[:user]) })
+    has_attachments(skip_render: lambda { |_, args| !can_view_solutions?(args[:user]) })
```

Comment:
Changed most if not all of the `if:` blocks that hid solutions to `skip_render:` blocks, because the `if:` blocks were blocking parsing as well as rendering.